File: segundo/1er cuatri/PDOO/ejercicios_clase/practica2_ejercicios_clase/practica2_ejercicios_clase_ruby/inventario.rb
# frozen_string_literal: true

class Inventario
  def initialize(id, capacidad)
    @id = id
    @capacidad = capacidad
    @objetos = Array.new
  end

  def to_s
    "Inventario ID: #{@id}, Capacidad: #{@capacidad} \n Objetos: #{@objetos}"
  end

  def add_objeto(obj)
    if @objetos.size < @capacidad
      @objetos.push(obj)
    else
      puts "lleno"
    end
  end

  def quitar_objeto(obj)
    @objetos.delete(obj)
  end
end

File: segundo/1er cuatri/PDOO/ejercicios_clase/practica2_ejercicios_clase/entregas_ejercicios/entrega_16-10/practica2_ejercicios_clase_java/src/practica2_ejercicios_clase/Inventario.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package practica2_ejercicios_clase;
import java.util.ArrayList;
/**
 *
 * @author alvarogz
 */
public class Inventario {
    private int id;
    private int capacidad;
    private ArrayList<Objeto> objetos;

    public Inventario(int id, int capacidad) {
        this.id = id;
        this.capacidad = capacidad;
        objetos = new ArrayList<>();
    }
    
    @Override
    public String toString() {
        return "Inventario ID: " + id + ", capacidad: " + capacidad + "\nobjetos: "+ objetos;
    }
    
    public void addObjeto(Objeto objeto) {
        if (objetos.size() < capacidad){
            objetos.add(objeto);
        }
        else
            System.out.println("Lleno");
    }
    
    public void quitarObjeto(Objeto objeto) {
        objetos.remove(objeto);
    }
            
}
